Web3Swift/RemoteProcedure/EstimateGasProcedure.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Created by Timofey on 3/14/18.
+//
+
+import Foundation
+import SwiftyJSON
+
+public final class EstimateGasProcedure: RemoteProcedure {
+
+    private let network: Network
+
+    private let senderAddress: BytesScalar
+    private let recipientAddress: BytesScalar
+    private let gasPrice: NumberScalar
+    private let weiAmount: NumberScalar
+
+    /**
+    Ctor
+
+    - parameters:
+        - network: network to call
+        - senderAddress: bytes representation of a sender address
+        - recipientAddress: bytes representation of a recipient address
+        - gasPrice: price that will be paid for each unit of gas
+        - weiAmount: amount to be sent from sender to recipient in wei
+    */
+    public init(
```

Comment:
@biboran Consider reducing the number of constructor parameters to 3 or less.

Reply:
@driver733 do you see any way to decompose/aggregate this class somehow?

Reply:
We will have even more parameters in other procedures. They depend on JSON-RPC implementation rather than on our domain logic.

Reply:
Event for gas estimation we will have two more parameters  `data`, `gas` ([ref.](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_call))

Reply:
@rockfridrich I think it will most probably be impossible to reasonably decompose `EstimateGasProcedure` but we can certainly think about some object that encapsulates sender and recipient although I don't see any need.

Reply:
@driver733 @rockfridrich lets carry on with the discussion in #91